storagemarket/lp2pimpl/net.go

```diff
@@ -146,3 +186,44 @@ func (p *DealProvider) handleNewDealStream(s network.Stream) {
 		return
 	}
 }
+
+func (p *DealProvider) handleNewDealStatusStream(s network.Stream) {
+	defer s.Close()
+
+	_ = s.SetReadDeadline(time.Now().Add(providerReadDeadline))
+	defer s.SetReadDeadline(time.Time{}) // nolint
+
+	var req types.DealStatusRequest
+	err := req.UnmarshalCBOR(s)
+	if err != nil {
+		log.Warnw("reading deal status request from stream", "err", err)
+		return
+	}
+	log.Debugw("received deal status request", "id", req.DealUUID, "client-peer", s.Conn().RemotePeer())
+
+	// Set a deadline on writing to the stream so it doesn't hang
+	_ = s.SetWriteDeadline(time.Now().Add(providerWriteDeadline))
+	defer s.SetWriteDeadline(time.Time{}) // nolint
+
+	pds, err := p.prov.Deal(p.ctx, req.DealUUID)
+	if err != nil && err == storagemarket.ErrDealNotFound {
```

Comment:
I suggest changing this to
```suggestion
	if err != nil && xerrors.Is(err, storagemarket.ErrDealNotFound) {
```
in case we change the Deal function to return something like `fmt.Errorf("some err: %w", ErrDealNotFound)`